### PR TITLE
feat: Compute entropy across full vocab for logging

### PR DESCRIPTION
**Pull request by parthchadha**
# What does this PR do ?

**Add a one line overview of what this PR aims to accomplish.**

# Issues
List issues that this PR closes ([syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)):

# Usage
* **You can potentially add a usage example below**

```python
# Add a code snippet demonstrating how to use this
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/NVIDIA-NeMo/RL/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you run the unit tests and functional tests locally? Visit our [Testing Guide](/NVIDIA-NeMo/RL/blob/main/docs/testing.md) for how to run tests
- [ ] Did you add or update any necessary documentation? Visit our [Document Development Guide](/NVIDIA-NeMo/RL/blob/main/docs/documentation.md) for how to write, build and test the docs.

# Additional Information
* ...

## Summary by CodeRabbit

* **New Features**
  * Adds accurate next-token entropy computation that works across tensor-parallel vocabulary shards.
  * Exposes a new metric, full_entropy, alongside the existing approximate entropy for comparison.
  * Improves numerical stability of entropy calculations, providing more reliable training metrics.

### CODE REVIEW

**Comment by ffrujeri** [comment]
Hey @parthchadha, in VeRL:
`
        curr_entropy = -torch.sum(curr_logprobs.exp() * curr_logprobs, dim=-1).unsqueeze(-1)
        entropy_loss = masked_mean(
            curr_entropy,
            mask,
            global_normalization_factor=global_valid_toks,
        )
`

**Review by ffrujeri — APPROVED** [review]

**Review by terrykong — COMMENTED** [review]
@parthchadha could you add a unit test?
